packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Utilities/FormattableStringHelpers.cs

```diff
@@ -97,5 +98,130 @@ public static string ReplaceLast(this string text, string oldValue, string newVa
+                if (isLiteral)
+                {
+                    var numSplits = span.Split(splitIndices, "\n");
+                    for (int i = 0; i < numSplits; i++)
+                    {
+                        var part = span[splitIndices[i]];
+                        // the literals could contain { and }, but they are unescaped. Since we are putting them back into the format, we need to escape them again.
+                        var startsWithCurlyBrace = part.Length > 0 && (part[0] == '{' || part[0] == '}');
+                        var start = startsWithCurlyBrace ? 1 : 0;
+                        var endsWithCurlyBrace = part.Length > 0 && (part[^1] == '{' || part[^1] == '}');
+                        var end = endsWithCurlyBrace ? part.Length - 1 : part.Length;
+                        if (startsWithCurlyBrace)
+                        {
+                            formatBuilder.Append(part[0]).Append(part[0]);
+                        }
+                        if (start <= end) // ensure that we have follow up characters before we move on
+                        {
+                            formatBuilder.Append(part[start..end]);
+                            if (endsWithCurlyBrace)
+                            {
+                                formatBuilder.Append(part[^1]).Append(part[^1]);
+                            }
+                        }
+                        if (i < numSplits - 1)
+                        {
+                            FormattableString formattableString = FormattableStringFactory.Create(formatBuilder.ToString(), args.ToArray());
+                            result.Add(formattableString);
+                            formatBuilder.Clear();
+                            args.Clear();
+                        }
+                    }
+                }
+                // if not Literal, is Args - recurse through Args and check if args has breaklines
+                else
+                {
+                    var arg = input.GetArgument(index);
+                    // when span has a format specifier, go into else case
+                    // if span contains ':' we don't need to split arg and add it directly to FormatBuilder
+                    // TODO: The following logic of the FormatSpecifier handling is temporary until we have a case where FormatSpecifier and \n both exist in an argument.
+                    // TODO: https://github.com/microsoft/typespec/issues/5255
+                    if (!span.Contains(':') && arg is string str)
+                    {
+                        ReadOnlySpan<char> strSpan = str.AsSpan();
+                        int start = 0, end = 0;
+                        bool isLast = false;
+                        // go into the loop when there are characters left
+                        while (end < strSpan.Length)
+                        {
+                            var indexOfLF = strSpan[start..].IndexOf('\n');
```

Comment:
Should we also handle \r\n? We wouldn't want different behavior for different on linux vs windows.

Reply:
I think we could just omit the `\r` and only handle the `\n` part.
How about this way:
we do the line breaks just like we have done in this PR, and then do a trim `\r` in the end (this could happen along with the separation of lines)

Reply:
looks like we need more test cases on `\r\n`, I will add them later

Reply:
I think it would be better to just handle both \n and \r\n so we don't unintentionally mangle the strings. If something just has \r by itself, we shouldn't trim it.